### PR TITLE
Transfer docs from cylc-flow repo.

### LINKED ISSUE #2: Transfer cylc user guide from cylc/cylc-flow
As discussed elsewhere, cylc-8 docs will need to span multiple components, not just cylc-flow, so best in a separate project.

### PR DESCRIPTION
**Pull request by hjoliver**
Close #2 

This takes the whole doc directory from cylc/cylc-flow, and relevant bits of setup.py and travis config ... which will probably require some tweaking, or maybe travis is not appropriate here at all given issues related to dependence on other components (below)?
 
**Questions**

- our docs are sphinx generated, which suggests that Python `setup.py` is still appropriate for this repo?

- but how to manage dependence on cylc/cylc-flow ? - needed to generate the command transcripts included in the user guide?
  - normal setup.py dependence requires uploading new versions cylc-flow before generating docs?
  - can we depend on local a local clone for building docs locally?
  - or not specify the dependence here formally, but manually ensure that we run this doc build within an environment that has access to the right cylc-flow version?

### CODE REVIEW

**Comment by oliver-sanders** [comment]
#### My Answers

> our docs are sphinx generated, which suggests that Python setup.py is still appropriate for this repo?

I don't think it matters much how we build the docs. The "standard" way to build Sphinx documentation is via the `Makefile` which Sphinx auto-generates when you start a project via `sphinx-quickstart`. Personally I would just make the minor modifications to the makefile as necessary.

> but how to manage dependence on cylc/cylc-flow

This repo must have a dependence on Cylc Flow in order to auto-document from docstrings etc.

I think the Cylc documentation should probably be versioned to the Cylc Flow version (see #17). We could just hardcode the Cylc Flow version in the `setup.py`.

The other alternative would be using the `cylc` namespace to deliver the "recommended" Cylc bundle (flow, UI-Serv, web-ui, etc)  and using this "bundle" as the documentation version...

On a side note we need to get rid of the command which documents the CLI (perhaps with https://github.com/cylc/cylc-flow/issues/2972)

**Comment by sadielbartholomew** [comment]
I started a PR to fix the build for the docs as on ``cylc-flow`` as it currently appears to be broken: cylc/cylc-flow#3179. However, I haven't had time to go back to it since. Is it still worth finishing or do we plan to get this merged soon, which will nullify the core of that PR so I can close it (or recast it, as there are some minor singular line changes in there that should still go be applicable)?

**Comment by hjoliver** [comment]
@sadielbartholomew - I'd like to get this merged soon.  We can tweak doc build and dependency specification later as needed (do you agree @oliver-sanders?)  For the moment we can put reminder Issues up on this repo, for stuff to do once the docs have moved here.

**Comment by sadielbartholomew** [comment]
Sure, thanks @hjoliver. I'll close that PR, in that case, & once the docs are moved over have a look to see what may still be relevant from it & open up a new PR on the docs repo if it seems suitable.

**Comment by oliver-sanders** [comment]
Seems fine to me, the packaging is broken, have put up a PR to fix:

https://github.com/hjoliver/cylc-doc/pull/1

File: bin/cylc-check-software [hunk 1]
```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env python3
```
Comment (kinow) [inline bin/cylc-check-software]: Out of curiosity, why did we bring just the `cylc-check-software` with this PR?

Comment (hjoliver) [reply bin/cylc-check-software]: On this branch, I stripped everything out of `cylc-check-software` except for the doc build requirements.  Not necessarily thinking we should keep it here, I was just trying to do a minimal split of our existing functionality as a first cut.
```
[oliverh@niwa-1007823l cylc-doc]$ bin/cylc-check-software 
Checking your software...

Individual results:
==========================================================================================
Package (version requirements)                                     Outcome (version found)
==========================================================================================
                                   *REQUIRED SOFTWARE*                                   
Python (3.7+).....................................FOUND & min. version MET (3.7.3.final.0)

                      *OPTIONAL SOFTWARE for the HTML documentation*                      
Python:sphinx (1.5.3+).......................................................NOT FOUND (-)
==========================================================================================

Summary:
                               ****************************                               
                                  Core requirements: ok                                  
                                Full-functionality: not ok                                
                               ****************************                               
```

**Comment by hjoliver** [comment]
Merged in @oliver-sanders PR.

File: bin/cylc-check-software (hunk 1 above)
Comment (hjoliver) [reply bin/cylc-check-software]: (And in the cylc-flow PR, I stripped the doc build bits out of `cylc-check-software`)

Comment (kinow) [reply bin/cylc-check-software]: :+1: got it! Makes sense now, thanks!

**Pull request merged**